src/Symfony/Component/HttpClient/Tests/CurlHttpClientTest.php

```diff
@@ -71,4 +71,49 @@ public function log($level, $message, array $context = [])
         ];
         $this->assertSame($expected, $logger->logs);
     }
+
+    public function testJsonRegexBuffer(): void
+    {
+        $client = new CurlHttpClient(['buffer' => '/([a-z]+\/json$)/']);
+        $response = $client->request('GET', 'https://jsonplaceholder.typicode.com/users');
```

Comment:
instead of hitting external domains we don't control, we should hit the test server (see existing test cases)
also since both clients have the same behavior, we should be able to move the test cases to the parent class

Reply:
I moved the tests to the base unit test class and changed the remote server. Thanks for pointing to this local test server